packages/stores/src/ui-config/convert-to-stake.ts

```diff
@@ -0,0 +1,205 @@
+import { CosmosQueries, IQueriesStore } from "@keplr-wallet/stores";
+import { CoinPretty, Dec, PricePretty, RatePretty } from "@keplr-wallet/unit";
+import { action, computed, makeObservable, observable } from "mobx";
+import { IPriceStore } from "src/price";
+
+import {
+  AccountStore,
+  CosmosAccount,
+  CosmwasmAccount,
+  OsmosisAccount,
+  OsmosisAccountImpl,
+} from "../account";
+import { DerivedDataStore } from "../derived-data";
+import { OsmosisQueries } from "../queries";
+import { QueriesExternalStore } from "../queries-external";
+
+export type SuggestedConvertToStakeAssets = {
+  poolId: string;
+  totalValue: PricePretty;
+  userPoolAssets: CoinPretty[];
+  currentApr: RatePretty;
+  sendConvertToStakeMsg: (newValidatorAddress?: string) => Promise<void>;
+};
+
+/** Aggregates user balancer shares are available to be staked, preferably for a higher APR.
+ *  Manages the state for the selection process, with selections keyed by pool ID.
+ *  Then, provides async callbacks for converting that stake. */
+export class UserConvertToStakeConfig {
+  @observable
+  protected _selectedConversionPoolId: string | null = null;
+
+  @computed
+  get selectedConversionPoolId() {
+    return this._selectedConversionPoolId;
+  }
+
+  @computed
+  get selectedConversion() {
+    return this.suggestedConvertibleAssetsPerPool.find(
+      ({ poolId }) => poolId === this.selectedConversionPoolId
+    );
+  }
+
+  /** For each user owned pool, this provides a breakdown of shares
+   *  that are of lesser returns than the staking inflationary returns.
+   *
+   *  Further, it provides a callback for converting those assets to stake via the
+   *  given account store. This callback expects a validator address if the user
+   *  has not yet selected a validator set preference. */
+  @computed
+  get suggestedConvertibleAssetsPerPool(): SuggestedConvertToStakeAssets[] {
```

Comment:
This function is rather long, can any of the code be extracted into helper functions to make it easier to read?

Reply:
What if I annotate chunks of code with high-level comments that don't repeat the code? Then, I can achieve the same effect without introducing new free-standing components to be reconciled.

Reply:
I'll leave it up to you. I think that annotations help, but in general whenever I see 100+ lines in a function, my eyes start to glaze over 😆